streams/src/test/java/org/apache/kafka/streams/processor/internals/StandbyTaskTest.java

```diff
@@ -323,8 +505,34 @@ public void shouldNotThrowUnsupportedOperationExceptionWhenInitializingStateStor
         final InternalTopologyBuilder internalTopologyBuilder = InternalStreamsBuilderTest.internalTopologyBuilder(builder);
         final ProcessorTopology topology = internalTopologyBuilder.setApplicationId(applicationId).build(0);
 
-        new StandbyTask(taskId, partitions, topology, consumer, changelogReader, config,
+        final StandbyTask standbyTask = new StandbyTask(taskId, partitions, topology, consumer, changelogReader, config,
+            new MockStreamsMetrics(new Metrics()), stateDirectory);
+
+        standbyTask.initializeStateStores();
+    }
+
+    @Test
+    public void shouldNotThrowUnsupportedOperationExceptionWhenInitializingWindowStores() throws IOException {
+        final String changelogName = "test-application-my-store-changelog";
+        final List<TopicPartition> partitions = Utils.mkList(new TopicPartition(changelogName, 0));
+        consumer.assign(partitions);
+        final Map<TopicPartition, OffsetAndMetadata> committedOffsets = new HashMap<>();
+        committedOffsets.put(new TopicPartition(changelogName, 0), new OffsetAndMetadata(0L));
+        consumer.commitSync(committedOffsets);
+
+        restoreStateConsumer.updatePartitions(changelogName, Utils.mkList(
+            new PartitionInfo(changelogName, 0, Node.noNode(), new Node[0], new Node[0])));
+        final InternalStreamsBuilder builder = new InternalStreamsBuilder(new InternalTopologyBuilder());
+        builder.stream(Collections.singleton("topic"), new ConsumedInternal<>()).groupByKey().windowedBy(TimeWindows.of(100)).count();
```

Comment:
as above. 

Btw: all code besides `windowedBy()` seems to be the same as above. Refactor into one `private` method to share code?